File: 준영/LV2/스킬트리.js
function solution(skill, skill_trees) {
  var answer = 0
  for (let i = 0; i < skill_trees.length; i++) {
    var idx = 0
    var loop = true
    for (const j of skill_trees[i]) {
      if (skill.includes(j)) {
        if (skill[idx] != j) {
          loop = false
          break
        } else {
          idx += 1
        }
      } 
    }  
    if (loop) {
      answer += 1
    }
  }
  return answer
}

skill = "CBD"
skill_trees = ["BACDE", "CBADF", "AECB", "BDA"]
solution(skill, skill_trees)

File: 준영/프로그래머스/LV2/스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    for st in skill_trees:
        idx = 0
        for i in st:
            if i in skill:
                if skill[idx] != i:
                    break
                else:
                    idx += 1
        else:
            answer += 1
        
    return answer


skill = "CBD"
skill_trees = ["BACDE", "CBADF", "AECB", "BDA"]
solution(skill, skill_trees)